solidity/app/contracts/Router.sol

```diff
@@ -114,38 +117,140 @@ abstract contract Router is AbacusConnectionClient, IMessageRecipient {
     }
 
     /**
-     * @notice Dispatches a message to an enrolled router via the local router's
-     * Outbox
-     * @dev Reverts if there is no enrolled router for _destination
-     * @param _destination The domain of the chain to which to send the message
-     * @param _msg The message to dispatch
+     * @notice Dispatches a message to an enrolled router via the local router's Outbox.
+     * @notice Does not pay interchain gas or create a checkpoint.
+     * @dev Reverts if there is no enrolled router for _destination.
+     * @param _destination The domain of the chain to which to send the message.
+     * @param _msg The message to dispatch.
      */
-    function _dispatchToRemoteRouter(uint32 _destination, bytes memory _msg)
+    function _dispatch(uint32 _destination, bytes memory _msg)
         internal
         returns (uint256)
     {
-        // ensure that destination chain has enrolled router
-        bytes32 _router = _mustHaveRemoteRouter(_destination);
-        return _outbox().dispatch(_destination, _router, _msg);
+        return _dispatch(_outbox(), _destination, _msg);
```

Comment:
TIL that solidity has function overloading. Is that a typical practice though?

Reply:
I haven't heard anything against it: https://docs.soliditylang.org/en/v0.8.12/contracts.html#function-overloading